File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int freq = 0,ans =0;

        for(int i=0;i<nums.size();i++){
            if(freq==0){
                ans = nums[i];
            }
            if(ans==nums[i]){
                freq++;
            }
            else{
                freq--;
            }
        }

        int count =0;
        for(int val: nums){
            if(val==ans){
                count++;
            }
        }

        if(count>nums.size()/2) return ans;
        else  return -1;

        return ans;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
    //     int candidate = nums[0];
    //     int count =1;
 
    //  for(int i=1;i<nums.length;i++){
    //     if(count==0){
    //         candidate = nums[i];
    //         count =1;
    //     }else if(nums[i] == candidate){
    //         count++;
    //     }else{
    //         count--;
    //     }

    //  }
    //  return candidate;

    HashMap<Integer,Integer> map = new HashMap<>();

    for(int num : nums){
        map.put(num,map.getOrDefault(num,0)+1);
        if(map.get(num) > nums.length/2){
            return num;
        }
    }
    return -1;


    }
}